pex/distribution_target.py

```diff
@@ -94,17 +94,21 @@ def requirement_applies(
         requirement,  # type: Requirement
         extras=None,  # type: Optional[Tuple[str, ...]]
     ):
-        # type: (...) -> bool
+        # type: (...) -> Optional[bool]
```

Comment:
This tri-state is needed to support build-time resolves for foreign platforms. We do not have information to build an environment markers dictionary without a foreign platform's target interpreter present. As such, we can only answer "Maybe" (None) to questions about environment marker applicability. We use this tristate to include distributions without checking environment markers when we can't when building a PEX (i.e.: we err on being overly inclusive). At runtime on the foreign platform, the resolve will be exact an winnow out any extra distributions we shipped over-eagerly. So we get a correct runtime resolve at the cost of potentially including more foreign platform distributions than needed. We can do no better here and this is a fairly narrow edge case.